packages/replay/src/integration.ts

```diff
@@ -339,6 +346,27 @@ Sentry.init({ replaysOnErrorSampleRate: ${errorSampleRate} })`,
       recordingOptions: this._recordingOptions,
     });
   }
+
+  /** Get canvas options from ReplayCanvas integration, if it is also added. */
+  private _maybeLoadFromReplayCanvasIntegration(): void {
+    // To save bundle size, we skip checking for stuff here
+    // and instead just try-catch everything - as generally this should all be defined
+    /* eslint-disable @typescript-eslint/no-non-null-assertion */
+    try {
+      const client = getClient()!;
+      const canvasIntegration = client.getIntegrationById!('ReplayCanvas') as Integration & {
+        getOptions(): Partial<ReplayConfiguration>;
+      };
+      if (!canvasIntegration) {
+        return;
+      }
+
+      this._replay!.addIntegration('canvas', canvasIntegration.getOptions());
```

Comment:
wdyt @mydea - added this method to keep the config out of `_experiments` - (this also breaks backwards compat, which I'm ok with because the low # of early adopters and to force them to use `ReplayCanvas` which is a much better interface anyway)

I wanted this out of `_experiments` so that it's not exposed to users

Reply:
I'm fine with this, IMHO we could also go with simply having a `replay.canvas` field and skip the integrations abstraction, but no strong feelings - depending on if we think we'll be adding some more integrations like this down the line.